connect/cli/plugins/project/extension_helpers.py

```diff
@@ -204,9 +204,37 @@ def _entrypoint_validations(project_dir, extension_dict):
     if errors:
         raise ClickException(f'Capability-Method errors: {errors}')
 
+    variables = ext_descriptor.get('variables', [])
+    errors = _have_variables_section_proper_shape(variables)
+    if errors:
+        raise ClickException(f'Variable errors: {errors}')
+
     _have_methods_proper_type(CustomExtension, capabilities)
 
 
+def _have_variables_section_proper_shape(variables):
+    if not variables:
+        return
+    if not isinstance(variables, list):
+        return 'The variables section should be a list of dictionaries.'
+
+    errors = []
+    checked = set()
+    for var in variables:
+        if not isinstance(var, dict):
+            errors.append(f'Variable {var} is not a dictionary.')
+        elif 'name' not in var.keys():
+            errors.append(f'Variable {var} does not have `name` key.')
+        else:
+            var_name = var['name'].lower()
+            if var_name not in checked:
+                checked.add(var_name)
+            else:
+                errors.append(f'Variable {var_name} is already defined on Variables section.')
```

Comment:
Variable {var_name} is duplicated on the variables definition in extension.json file

Reply:
Sorry, but I don't understand what do you mean...
Could you explain a bit more in detail?

Reply:
Understood....I'm gonna change the error message with your suggestion...thnx!